docs/topics/tour/kotlin-tour-intermediate-functions.md

```diff
@@ -0,0 +1,935 @@
+But what if you want to use the returned object later on in your code? You could achieve this with an if condition **and** an 
+else branch:
+
+```kotlin
+fun sendNotification(recipientAddress: String): String {
+    println("Yo $recipientAddress!")
+    return "Notification sent!"
+}
+
+fun getNextAddress(): String {
+    return "sebastian@jetbrains.com"
+}
+
+fun main() { 
+    //sampleStart
+    val addr: String? = getNextAddress()
+    val confirm = if(addr != null) {
+        sendNotification(addr)
+    } else { null }
+    //sampleEnd
+}
+```
+{kotlin-runnable="true" id="kotlin-tour-scope-function-let-non-null-if"}
+
+However, a more concise approach is to use the `.let()` scope function:
+
+```kotlin
+fun sendNotification(recipientAddress: String): String {
+    println("Yo $recipientAddress!")
+    return "Notification sent!"
+}
+
+fun getNextAddress(): String {
+    return "sebastian@jetbrains.com"
+}
+
+fun main() {
+    //sampleStart
+    val addr: String? = getNextAddress()
+    val confirm = addr?.let {
+        sendNotification(it)
+    }
+    //sampleEnd
+}
+```
+{kotlin-runnable="true" id="kotlin-tour-scope-function-let-non-null"}
+
+The example:
+* Creates a variable called `confirm`.
+* Uses a safe call to call the `.let()` scope function on the `addr` variable.
+* Creates a temporary scope within the `.let()` scope function so that you don't have to explicitly refer to the `addr` variable.
+* Passes the `sendNotification()` function as a lambda expression into the `.let()` scope function.
+* Uses the temporary scope to refer to the `addr` variable via `it`.
+* Assigns the result to the `confirm` variable.
+
+With this approach, your code can handle the `addr` variable potentially being a `null` value, and you can use the 
+`confirm` variable later on in your code.
+
+### Apply
+
+Use the `.apply()` scope function to initialize objects, like a class instance, at the time of creation rather than later
+on in your code. This approach makes your code easier to read and manage.
+
+Consider the example:
+
+```kotlin
+class Client() {
+    var token: String? = null
+    fun connect() = println("connected!")
+    fun authenticate() = println("authenticated!")
+    fun getData(): String = "Mock data"
+}
+
+val client = Client()
+
+fun main() {
+    client.token = "asdf"
+    client.connect()
+    // connected!
+    client.authenticate()
+    // authenticated!
+    client.getData()
+}
+```
+{kotlin-runnable="true" id="kotlin-tour-scope-function-apply-before"}
+
+The example has a `Client` class that contains one property called `token` and three member functions: `connect()`,
+`authenticate()`, and `getData()`.
+
+The example creates `client` as an instance of the `Client` class before initializing its `token` property and calling its
+member functions in the `main()` function.
+
+Although this example is compact, in the real world it can be a while after the creation of your class instance before you
```

Comment:
I'm trying to emphasize that it can be a while between the instance creation and the configuration of it. I think your suggested version makes that less clear so I'll keep the original :)

Reply:
I get that, it's just the word order is a bit entangled for me. Another option:

> Although this example is compact, in the real world, it can take a while before you can configure and use the class instance (and its member functions) after you've created it.